src/components/QRShare/index.js

```diff
@@ -0,0 +1,98 @@
+import React, {Component} from 'react';
+import QRCodeLibrary from 'react-native-qrcode-svg';
+import {View} from 'react-native';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import defaultTheme from '../../styles/themes/default';
+import styles from '../../styles/styles';
+import Text from '../Text';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import compose from '../../libs/compose';
+import variables from '../../styles/variables';
+import ExpensifyWordmark from '../../../assets/images/expensify-wordmark.svg';
+import {qrSharePropTypes, qrShareDefaultProps} from './propTypes'
+
+const propTypes = {
+    ...qrSharePropTypes,
+    ...windowDimensionsPropTypes,
+    ...withLocalizePropTypes,
+};
+
+class QRShare extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            qrCodeSize: 0,
+        };
+
+        this.onLayout = this.onLayout.bind(this);
+        this.getSvg = this.getSvg.bind(this);
+    }
+
+    onLayout(event) {
+        this.setState({
+            qrCodeSize: event.nativeEvent.layout.width - variables.qrShareHorizontalPadding * 2,
+        });
+    }
+
+    getSvg() {
+        return this.svg;
+    }
+
+    render() {
+        return (
+            <View
+                style={styles.shareCodeContainer}
+                onLayout={this.onLayout}
+            >
+                <View
+                    style={{
+                        alignSelf: 'stretch',
+                        height: 27,
+                        marginBottom: 20,
+                    }}
```

Comment:
Hi everyone, coming from a follow-up PR for this component. Just wanted to note that as per our [styling guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/STYLING.md#inline-styles), let's remember to avoid inline styles. Thanks!

Reply:
got it, thanks for the feedback! 👍